packages/hawtio/src/plugins/jmx/JmxTreeView.tsx

```diff
@@ -6,23 +6,54 @@ import { MBeanTreeContext } from './context'
 import './JmxTreeView.css'
 
 export const JmxTreeView: React.FunctionComponent = () => {
-  const { tree, setNode } = useContext(MBeanTreeContext)
-  const [expanded, setExpanded] = useState(false)
+  const { tree, node, setNode } = useContext(MBeanTreeContext);
+  const [expanded, setExpanded] = useState(false);
+  const [filteredTree, updateTree] = useState(tree.getTree());
 
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const onSearch = (event: ChangeEvent<HTMLInputElement>) => {
-    // TODO
+    if(!event.target.value) updateTree(tree.getTree());
+    
+    const treeElements = lookupSearchInTree(event.target.value, tree.getTree());
+
+    if (treeElements.length == 0) updateTree(tree.getTree())
+    else updateTree(treeElements);
   }
 
   const onSelect = (event: React.MouseEvent<Element, MouseEvent>, item: TreeViewDataItem) => {
     setNode(item as MBeanNode)
   }
 
+  const lookupSearchInTree = (search: string, tree?: MBeanNode[]) : MBeanNode[] => {
+    if (tree?.length == 0) return [];
+    
+    let results : MBeanNode[] = [];
+
+    for (let parentNode of tree || []) {
```

Comment:
This should be `const`. Also, do we need `|| []` part?  As emptiness of the list should be checked at the beginning of the function.

Reply:
Yes, if we remove the || [] part TS will throw an error stating that parentNode.children is of type MBeanNode | undefined, even though we are checking if it's null or not at the top.

Reply:
Actually the above comment is wrong. You were right that I wasn't checking if tree existed.